Unstable test `TestLocal`

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

Run `TestLocal`

### 2. What did you expect to see? (Required)

Test pass

### 3. What did you see instead (Required)

```
panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0xb1c0f5]



goroutine 168 [running]:

github.com/pingcap/tiproxy/pkg/balance/observer.TestLocal.func1()

	/home/jenkins/agent/workspace/pingcap/tiproxy/ghpr_unit_test/tiproxy/pkg/balance/observer/backend_observer_test.go:210 +0x215

github.com/stretchr/testify/assert.Eventually.func1()

	/go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:1852 +0x22

created by github.com/stretchr/testify/assert.Eventually in goroutine 227

	/go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:1852 +0x21e

FAIL	github.com/pingcap/tiproxy/pkg/balance/observer	2.334s
```

### 4. What is your version? (Required)

<!-- Paste the output of tiproxy version -->
<!-- Paste the output of tiproxyctl version if related -->
<!-- Paste the output of SELECT tidb_version()  if related -->

master

--- a/pkg/balance/observer/backend_observer_test.go
+++ b/pkg/balance/observer/backend_observer_test.go
@@ -206,8 +206,8 @@ func TestLocal(t *testing.T) {
 		require.Eventually(ts.t, func() bool {
 			result := ts.getResultFromCh()
 			require.NoError(ts.t, result.Error())
-			health := result.Backends()[backend]
-			return health.Local == test.local
+			health, ok := result.Backends()[backend]
+			return ok && health.Local == test.local
 		}, 3*time.Second, 10*time.Millisecond, "test case %d", i)
 		ts.removeBackend(backend)
 	}

